docs/openapi-docs/src/main/scala/sttp/tapir/docs/openapi/schema/ObjectSchemasForEndpoints.scala

```diff
@@ -82,14 +81,24 @@ object ObjectSchemasForEndpoints {
 
   private def objectSchemas(typeData: TypeData[_]): List[ObjectTypeData] = {
     typeData match {
-      case TypeData(s @ TSchema(st: TSchemaType.SProduct, _, _, _), validator) =>
-        List(st.info -> TypeData(s, validator): ObjectTypeData) ++ fieldsSchemaWithValidator(st, validator)
-          .flatMap(objectSchemas)
-          .toList
       case TypeData(TSchema(TSchemaType.SArray(o), _, _, _), validator) =>
         objectSchemas(TypeData(o, elementValidator(validator)))
-      case TypeData(s @ TSchema(st: TSchemaType.SCoproduct, _, _, _), validator) =>
-        (st.info -> TypeData(s, validator): ObjectTypeData) +: st.schemas.flatMap(c => objectSchemas(TypeData(c, Validator.pass)))
+      case TypeData(s @ TSchema(st: TSchemaType.SProduct, _, _, _), validator: Validator.Product[_]) =>
+        (st.info -> TypeData(s, validator): ObjectTypeData) +: fieldsSchemaWithValidator(st, validator)
+          .flatMap(objectSchemas)
+          .toList
+      case TypeData(s @ TSchema(st: TSchemaType.SProduct, _, _, _), Validator.CollectionElements(ev: Validator.Product[_], _)) =>
+        (st.info -> TypeData(s, ev: Validator[_]): ObjectTypeData) +: fieldsSchemaWithValidator(st, ev)
+          .flatMap(objectSchemas)
+          .toList
+      case TypeData(s @ TSchema(st: TSchemaType.SCoproduct, _, _, _), validator: Validator.Coproduct[_]) =>
+        (st.info -> TypeData(s, validator): ObjectTypeData) +: subtypesSchemaWithValidator(st, validator)
+          .flatMap(objectSchemas)
+          .toList
+      case TypeData(s @ TSchema(st: TSchemaType.SCoproduct, _, _, _), v @ Validator.CollectionElements(ev: Validator.Coproduct[_], _)) =>
+        (st.info -> TypeData(s, v: Validator[_]): ObjectTypeData) +: subtypesSchemaWithValidator(st, ev)
```

Comment:
I think that it should be like this:
```suggestion
        (st.info -> TypeData(s, ev: Validator[_]): ObjectTypeData) +: subtypesSchemaWithValidator(st, ev)
```
But I don't know how to prove it (i.e.  write test)

Reply:
there would need to be a global validator for the top-level coproduct type? not only for elements?

Reply:
I don't know. That's tricky question as by having a separate validator for top-level coproduct type we would imply that we are performing some merges between top-level coproduct validator and sub-validators.

Reply:
I suppose you could add a validator to the trait, with children also having individual validators? Then in the test, you would need an optional body with that trait, no?